aerospike/tests/test_aerospike.py

```diff
@@ -66,6 +69,34 @@ def test_e2e(dd_agent_check, instance):
     aggregator.assert_metrics_using_metadata(get_metadata_metrics())
 
 
+@requires_py3
+@pytest.mark.e2e
+def test_openmetrics_e2e(dd_agent_check, instance_openmetrics_v2):
+    version_parts = [int(p) for p in VERSION.split('.')]
+
+    aggregator = dd_agent_check(instance_openmetrics_v2, rate=True)
+
+    for metric in EXPECTED_PROMETHEUS_METRICS:
+        aggregator.assert_metric(metric)
+
+    if version_parts >= [5, 6]:
+        for metric in EXPECTED_PROMETHEUS_METRICS_5_6:
+            aggregator.assert_metric(metric)
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics(), check_submission_type=True)
+
+
+@requires_py3
+@pytest.mark.integration
+def test_metrics_warning(dd_run_check, instance_openmetrics_v2):
```

Comment:
What is this test checking? Is there a more detailed Exception that we can catch? Is there a service check we can collect even if it fails?

Reply:
Hi @yzhan289 this test checks that the check raises an exception if the legacy `metrics` parameter is present alongside `openmetrics_endpoint` to test the validator in [validators.py](https://github.com/DataDog/integrations-core/blob/d15349bab45defe95c47a8bcd8a66a642fbda2b0/aerospike/datadog_checks/aerospike/config_models/validators.py). I just added a `match` argument to make that clearer and assert an error message- let me know what you think!